### PR TITLE
Added description of opening dialogs programatically

### PR DESCRIPTION
**Pull request by piotr-sobczyk**
Jira task: https://tipser.atlassian.net/browse/T3-4340

Added following sub-section to docs (inside a new "advanced usage" section):

image

### CODE REVIEW

File: source/includes/_advanced-usage.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#Advanced usage
+
+## Opening product dialog programatically
+
+Tipser product dialog can be openend programatically by using the underlying [Tipser SDK](#tipser-sdk) instance.
```
Comment (mateuszsikora) [inline source/includes/_advanced-usage.md]: afaik only old dialog you can open in this way :(

Comment (chmurson) [reply source/includes/_advanced-usage.md]: That's actually very fair point, that getting rid of old product dialog is more complicated than we thought so initially.

File: source/includes/_advanced-usage.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+1. Create a custom React component and mount it anywhere under `TipserElementsProvider` in React elements hierarchy.
+2. Inside that component you can use tipserSdk instance from the context `TipserContext`. This how it can be done with [React Hooks](https://reactjs.org/docs/hooks-intro.html):
+
+
+```javascript
+import { TipserContext } from `@tipser/tipser-elements`;
+
+const DialogOpener = () => {
+  const context = useContext(TipserContext());
+  conts sdk = context.data.tipserSdk;
```
Comment (chmurson) [inline source/includes/_advanced-usage.md]:
```suggestion
  const sdk = context.data.tipserSdk;
```

File: source/includes/_advanced-usage.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  
+  useEffect(() => { sdk.openProductDialog(productId); }, []);
```
Comment (chmurson) [inline source/includes/_advanced-usage.md]:
```suggestion
  const productId = "5d0245b09bee660001745f07";
  useEffect(() => { sdk.openProductDialog(productId); }, []);
```

Comment (chmurson) [reply source/includes/_advanced-usage.md]: It's a bit off the good practices of react hooks - lack of `productId` in array but I guess that's okay;

I would prefer not to have `productId` coming from the void. It always struck me "where I should get it from?" when I see something like that in other products docs.

File: source/includes/_advanced-usage.md (hunk 1 above)
Comment (piotr-sobczyk) [reply source/includes/_advanced-usage.md]: @mateuszsikora you are right. However the old dialog is so far the only official one and customers are not aware of the new dialog. When the new dialog will become the official one, I will add clarifications to this section.

File: source/includes/_advanced-usage.md (hunk 3 above)
Comment (piotr-sobczyk) [reply source/includes/_advanced-usage.md]: I think the names like `posIId`, `config`, `productId` are pretty self descriptive and the customers who know our library enough to reach this far corner of our docs will have no trouble understanding it. However I will add a short sentence explaining how to get product ids. Is it okay?

Comment (piotr-sobczyk) [reply source/includes/_advanced-usage.md]: > It's a bit off the good practices of react hooks - lack of productId in array but I guess that's okay

Overall you are of course right. But in this specific example I think we just want to open a dialog once, no matter what (we don't want our component to keep opening dialogs if `productId` gets changed for some unknown reasons?)

Comment (piotr-sobczyk) [reply source/includes/_advanced-usage.md]: @chmurson does it look clearer now?

image

File: source/includes/_advanced-usage.md (hunk 1 above)
Comment (mateuszsikora) [reply source/includes/_advanced-usage.md]: :+1:

**Pull request merged**